app/lanes_test.go

```diff
@@ -0,0 +1,2160 @@
+
+	// Generate feeds transaction
+	feedsTxBytes := bandtesting.GenSequenceOfTxs(
+		s.txConfig.TxEncoder(),
+		s.txConfig,
+		[]sdk.Msg{
+			GenMsgSubmitSignalPrices(
+				&s.valAccWithNumSeq,
+				s.app.FeedsKeeper.GetCurrentFeeds(s.ctx).Feeds,
+				s.ctx.BlockTime().Unix(),
+			),
+		},
+		&s.valAccWithNumSeq,
+		sdk.Coins{},
+		1000000,
+		40,
+	)
+
+	s.checkTxAcceptance(feedsTxBytes, "feedsLane")
+
+	require.Equal(s.app.Mempool().CountTx(), 43)
+	mempool := s.app.Mempool().(*mempool.Mempool)
+	require.Equal(mempool.GetLane("defaultLane").CountTx(), 3)
+	require.Equal(mempool.GetLane("feedsLane").CountTx(), 40)
+
+	// Prepare proposal
+	prepareReq := &abci.RequestPrepareProposal{
+		MaxTxBytes:      1000000,
+		Height:          s.app.LastBlockHeight() + 1,
+		Time:            s.ctx.BlockTime(),
+		ProposerAddress: bandtesting.Validators[0].Address.Bytes(),
+	}
+
+	resp, err := s.app.PrepareProposal(prepareReq)
+	require.NoError(err)
+	require.NotNil(resp)
+	require.Equal(len(resp.Txs), 42)
+
+	var expectedTxBytes [][]byte
+	expectedTxBytes = append(append(append(append(expectedTxBytes, feedsTxBytes[:25]...), bankSendTxBytes[0]), feedsTxBytes[25:]...), bankSendTxBytes[1])
+
+	require.Equal(resp.Txs, expectedTxBytes)
+}
+
+// TestLargeTxSizeBlocksSubsequentTx tests that a large transaction size blocks subsequent transactions
+func (s *AppTestSuite) TestLargeTxSizeBlocksSubsequentTx() {
+	require := s.Require()
+
+	// Generate large bank send transaction
+	largeBankSendTxBytes := bandtesting.GenSequenceOfTxs(
+		s.txConfig.TxEncoder(),
+		s.txConfig,
+		[]sdk.Msg{
+			genMsgSend(&s.valAccWithNumSeq, &s.reporterAccWithNumSeq, sdk.Coins{sdk.NewInt64Coin("uband", 1)}),
+		},
+		&s.valAccWithNumSeq,
+		sdk.Coins{sdk.NewInt64Coin("uband", 12500)},
+		5000000,
+		1,
+	)
+
+	s.checkTxAcceptance(largeBankSendTxBytes, "defaultLane")
+
+	// Generate small bank send transaction
+	smallBankSendTxBytes := bandtesting.GenSequenceOfTxs(
+		s.txConfig.TxEncoder(),
+		s.txConfig,
+		[]sdk.Msg{
+			genMsgSend(&s.valAccWithNumSeq, &s.reporterAccWithNumSeq, sdk.Coins{sdk.NewInt64Coin("uband", 1)}),
+		},
+		&s.valAccWithNumSeq,
+		sdk.Coins{sdk.NewInt64Coin("uband", 250)},
+		100000,
+		1,
+	)
+
+	s.checkTxAcceptance(smallBankSendTxBytes, "defaultLane")
+
+	// Generate report data transaction
+	reportTxBytes := bandtesting.GenSequenceOfTxs(
+		s.txConfig.TxEncoder(),
+		s.txConfig,
+		[]sdk.Msg{
+			genMsgReportData(&s.valAccWithNumSeq),
+		},
+		&s.valAccWithNumSeq,
+		sdk.Coins{},
+		5000000,
+		2,
+	)
+
+	// Check that the report data transactions are accepted
+	s.checkTxAcceptance(reportTxBytes, "oracleReportLane")
+
+	// Generate tss transaction
+	tssTxBytes := bandtesting.GenSequenceOfTxs(
+		s.txConfig.TxEncoder(),
+		s.txConfig,
+		[]sdk.Msg{
+			genMsgSubmitDKGRound1(
+				&s.tssAccountsWithNumSeq[0],
+				s.tssGroupCtx.GroupID,
+				tss.MemberID(1),
+				s.tssGroupCtx.Round1Infos[0],
+			),
+		},
+		&s.tssAccountsWithNumSeq[0],
+		sdk.Coins{},
+		5000000,
```

Comment:
not sure it's possible to have some shared variable. the test is quite fix with duplicated hardcode. so when we change percentage of limit it affect multiple test cases.

or maybe we just override the percentage for test only so it's the fixed value. (not rely on value from app, only logic)

Reply:
I understand but I want them to be identical to the real app as much as possible